core/adapters/eth_tx.go

```diff
@@ -97,9 +94,7 @@ func createTxRunResult(
 		gasLimit,
 	)
 	if IsClientRetriable(err) {
-		var output models.RunResult
-		output.MarkPendingConnection()
-		return output
+		return models.RunResultPendingConnection()
```

Comment:
Unrelated to PR.  What do you think about renaming these functions to 'NewRun...()' it would clarify they are returning a new struct as opposed to running an action.

Reply:
I think it's a good idea, I will spin this off into a separate PR.